lib/rester.rb

```diff
@@ -19,7 +29,43 @@ def load_tasks
     def connect(service, params={})
       adapter_opts = Client::Adapters.extract_opts(params)
       adapter = Client::Adapters.connect(service, adapter_opts)
-      Client.new(adapter, params)
+      Client.new(adapter, params).tap { |client| adapter.get('/ping') }
+    end
+
+    def service_name
+      @_service_name ||= _get_service_name
+    end
+
+    def service_name=(name)
+      @_service_name = name
+    end
+
+    def correlation_id
+      _correlation_ids[Thread.current.object_id] ||= SecureRandom.uuid
+    end
+
+    def correlation_id=(id)
+      if id.nil?
+        _correlation_ids.delete(Thread.current.object_id)
+      else
+        _correlation_ids[Thread.current.object_id] = id
+      end
+    end
+
+    private
+
+    def _get_service_name
+      if defined?(Rails) && Rails
+        Rails.application.class.parent_name
+      else
+        services = Service.descendants
+        raise "Define a service name" if services.empty?
+        services.first.service_name
+      end
```

Comment:
I guess now that we have the automatic `ping` upon connection. This will happen immediately